pygbm/gradient_boosting.py

```diff
@@ -285,16 +366,33 @@ def __init__(self, loss='binary_crossentropy', learning_rate=0.1,
             verbose=verbose, random_state=random_state)
 
     def predict(self, X):
+        # This could be done in parallel
         return np.argmax(self.predict_proba(X), axis=1)
 
     def predict_proba(self, X):
         raw_predictions = self._raw_predict(X)
         return self.loss_.predict_proba(raw_predictions)
 
+    def _get_n_trees_per_iteration(self, y):
+        n_classes = np.unique(y).shape[0]
```

Comment:
Instead of `np.unique(y)` we should be using the `LabelEncoder` of scikit-learn to set the `classes_` attribute to allow for string class labels or non-contiguous integer class labels. Then instead of storing  of `n_trees_per_iteration_` attribute we could do:

```python
n_trees_per_iteration = 1 if self.classes_.shape[0] == 2 else self.classes_.shape[0]
```

where we actually need it.

Reply:
Ok I'll do that.

But we need `n_trees_per_iteration_` even in regression so I doubt we could get rid of the attribute. `GradientBoostingRegressor` has `_get_n_tree_per_iteration` that always returns `1`.

Reply:
Ok as you prefer.

Reply:
I mean of course I'm open to any other suggestions. I just don't think we can easily avoid having `n_trees_per_iterations_`.

But sure I'll use `LabelEncoder` to have better support for classification